src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -271,8 +271,14 @@ private void ReadSettings(PackageRestoreInputs packageRestoreInputs)
                     GetInstalledPackageReferences(packageReferenceFile, allowDuplicatePackageIds: true));
             }
 
+            // If PackageSaveMode.None is provided, this means that the user did not specify a
+            // PackageSaveMode. In the case of V2 restore, this should default to PackageSaveMode.Defaultv2.
+            var packageSaveMode = EffectivePackageSaveMode == Packaging.PackageSaveMode.None ?
```

Comment:
Still I think this can be solved by passing a parameter to the method, saying calculate effective mode for v2 or v3. But I won't insist.